src/algebra/Module/basic.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2019 Robert A. Spencer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert A. Spencer
+-/
+import algebra.module
+import algebra.punit_instances
+import category_theory.types
+import linear_algebra.basic
+open category_theory
+
+universe u
+
+variables (R : Type u) [ring R]
+
+/-- The category of R-modules and their morphisms. -/
+structure Module :=
+(carrier : Type u)
+[is_add_comm_group : add_comm_group carrier]
+[is_module : module R carrier]
+
+attribute [instance] Module.is_add_comm_group Module.is_module
+
+namespace Module
+
+instance : has_coe_to_sort (Module R) :=
+{ S := Type u, coe := Module.carrier }
+
+def of (X : Type u) [add_comm_group X] [module R X] : Module R := ⟨R, X⟩
+
+instance : has_zero (Module R) := ⟨ of R punit ⟩
```

Comment:
```suggestion
-- TODO: replace this with has_zero_object once the PR on zero objects is merged.
instance : has_zero (Module R) := ⟨of R punit⟩
```

Reply:
Do you have a reference for that PR?

Reply:
Voila: https://github.com/leanprover-community/mathlib/pull/1445/files

Reply:
Done.